docs/design/2020-09-12-utf8mb4_zh_0900_as_cs.md

```diff
@@ -0,0 +1,86 @@
+# Proposal: support `pinyin` order for `utf8mb4` charset
+
+- Author(s):     [xiongjiwei](https://github.com/xiongjiwei)
+- Last updated:  2020-09-15
+- Discussion at: https://github.com/pingcap/tidb/issues/19747
+
+## Abstract
+This proposal proposes a new feature that supports `pinyin` order for chinese character.
+
+## Background
+It's unable now to order by a column based on it's pinyin order. For example:
+
+```sql
+create table t(
+	a varchar(100)
+)
+charset = 'utf8mb4' collate = 'utf8mb4_zh_0900_as_cs';
+
+# insert some data:
+insert into t values ("中文"), ("啊中文");
+
+# a query requires to order by column a in its pinyin order:
+select * from t order by a;
++-----------+
+| a         |
++-----------+
+| 啊中文    |
+| 中文      |
++-----------+
+2 rows in set (0.00 sec)
+```
+
+## Proposal
+
+`pinyin` order for Chinese character supported by this proposal will add a new collation named `utf8mb4_general_zh_cs` which are exactly same with `gbk_bin`. Collation `utf8mb4_general_zh_cs` is for charset `utf8mb4`.
+
+Following SQL statements should have same result.
+```sql
+# order
+select * from t order by a collate utf8mb4_general_zh_cs;
+select * from t order by convert(a using gbk) collate gbk_bin;
+
+# sort key
+select weight_string(a collate utf8mb4_general_zh_cs);
+select weight_string(convert(a using gbk) collate gbk_bin);
+```
+
+## Rationale
+
+### How to implement
+
+Collation `utf8mb4_general_zh_cs` actually convert `utf8mb4` to `gbk` code point and do same thing with collation `gbk_bin`. For the compatibility with MySQL, our convert step should exactly same as MySQL `convert(... using gbk)`.
```

Comment:
Are you sure that it is appliable to use 'code point' as the weight string? Considering:

- The range of GBK 'code point'.
- Do you plan to use code point for all characters, or just Chinese characters?

If they're confirmed, please add more details about them in this proposal, thank you!

Reply:
after read source code and test some case, I found `gbk_bin` does represent `pinyin` order, and the default collation of `gbk` is `gbk_chinese_ci`. I had changed `utf8mb4_general_zh_cs` to `utf8mb4_general_zh_ci`. I had updated this proposal. PTAL